File: src/main/resources/statics/js/modules/unimall/unimallcoupon.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'unimall/unimallcoupon/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '代金券名称', name: 'title', index: 'title', width: 80 }, 			
			{ label: '使用类型，如满减', name: 'type', index: 'type', width: 80 }, 			
			{ label: '描述', name: 'description', index: 'description', width: 80 }, 			
			{ label: '', name: 'total', index: 'total', width: 80 }, 			
			{ label: '会员类型0:非会员1:会员2:全部', name: 'surplus', index: 'surplus', width: 80 }, 			
			{ label: '', name: 'limit', index: 'limit', width: 80 }, 			
			{ label: '减少金额', name: 'discount', index: 'discount', width: 80 }, 			
			{ label: '最低消费金额', name: 'min', index: 'min', width: 80 }, 			
			{ label: '是否可用 0不用 1可用', name: 'status', index: 'status', width: 80 }, 			
			{ label: '', name: 'categoryId', index: 'category_id', width: 80 }, 			
			{ label: '过期天数', name: 'days', index: 'days', width: 80 }, 			
			{ label: '领取开始时间', name: 'gmtStart', index: 'gmt_start', width: 80 }, 			
			{ label: '领取/使用结束时间', name: 'gmtEnd', index: 'gmt_end', width: 80 }, 			
			{ label: '', name: 'gmtUpdate', index: 'gmt_update', width: 80 }, 			
			{ label: '', name: 'gmtCreate', index: 'gmt_create', width: 80 }, 			
			{ label: '商户id', name: 'supplierId', index: 'supplier_id', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		unimallCoupon: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.unimallCoupon = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.unimallCoupon.id == null ? "unimall/unimallcoupon/save" : "unimall/unimallcoupon/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.unimallCoupon),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "unimall/unimallcoupon/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "unimall/unimallcoupon/info/"+id, function(r){
                vm.unimallCoupon = r.unimallCoupon;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});

File: src/main/java/com/icloud/modules/unimall/entity/UnimallCoupon.java
package com.icloud.modules.unimall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-05-16 10:38:18
 */
@Data
@TableName("t_unimall_coupon")
public class UnimallCoupon implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 代金券名称 */
       @TableField("title")
       private String title;
   	   	   /* 使用类型，如满减 */
       @TableField("type")
       private Integer type;
   	   	   /* 描述 */
       @TableField("description")
       private String description;
   	   	   /*  */
       @TableField("total")
       private Integer total;
   	   	   /* 会员类型0:非会员1:会员2:全部 */
       @TableField("surplus")
       private Integer surplus;
   	   	   /*  */
       @TableField("limit")
       private Integer limit;
   	   	   /* 减少金额 */
       @TableField("discount")
       private Integer discount;
   	   	   /* 最低消费金额 */
       @TableField("min")
       private Integer min;
   	   	   /* 是否可用 0不用 1可用 */
       @TableField("status")
       private Integer status;
   	   	   /*  */
       @TableField("category_id")
       private Long categoryId;
   	   	   /* 过期天数 */
       @TableField("days")
       private Integer days;
   	   	   /* 领取开始时间 */
       @TableField("gmt_start")
       private Date gmtStart;
   	   	   /* 领取/使用结束时间 */
       @TableField("gmt_end")
       private Date gmtEnd;
   	   	   /*  */
       @TableField("gmt_update")
       private Date gmtUpdate;
   	   	   /*  */
       @TableField("gmt_create")
       private Date gmtCreate;
   	   	   /* 商户id */
       @TableField("supplier_id")
       private Long supplierId;
   	
}
